pkg/loader/compose/v1v2.go

```diff
@@ -268,12 +269,154 @@ func libComposeToKomposeMapping(composeObject *project.Project) (kobject.Kompose
+func handleVolume(komposeObject *kobject.KomposeObject) {
+	for name, _ := range komposeObject.ServiceConfigs {
+		// retrieve volumes of service
+		vols := retrieveVolume(name, *komposeObject)
+		var FinalVol []kobject.Volumes
+
+		// Check for duplication of volumes
+		for _, v := range vols {
+			if !findDuplicateVol(v, FinalVol) {
+				FinalVol = append(FinalVol, v)
+			}
+		}
+		var count int
+
+		// Parse volume information and store it in Volumes struct
+		for v, _ := range FinalVol {
+			vol := &FinalVol[v]
+			var err error
+			vol.VolumeName, vol.Host, vol.Container, vol.Mode, err = transformer.ParseVolume(vol.MountPath)
+			if err != nil {
+				log.Warningf("Failed to configure container volume: %v", err)
+				continue
+			}
+			log.Debug("Volume name %s", vol.VolumeName)
+
+			if vol.VFrom == "" {
+				vol.PVCName = fmt.Sprintf("%s-claim%d", vol.SvcName, count)
+				count++
+			}
+
+		}
+
+		// We can't assign value to struct field in map while iterating over it, so temporary variable `temp` is used here
+
+		var temp = komposeObject.ServiceConfigs[name]
+		temp.Volumes = FinalVol
+		komposeObject.ServiceConfigs[name] = temp
+		for _, service := range komposeObject.ServiceConfigs {
+			for v, _ := range service.Volumes {
+				vol := &service.Volumes[v]
+				if vol.VFrom != "" {
+					for _, j := range komposeObject.ServiceConfigs[vol.VFrom].Volumes {
+						if vol.MountPath == j.MountPath {
+							vol.PVCName = j.PVCName
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
 func checkLabelsPorts(noOfPort int, labels string, svcName string) error {
 	if noOfPort == 0 && (labels == "NodePort" || labels == "LoadBalancer") {
 		return errors.Errorf("%s defined in service %s with no ports present. Issues may occur when bringing up artifacts.", labels, svcName)
 	}
 	return nil
 }
+
+// returns true if duplicate volumes are there
+// it's possible when `volumes_from` key is used
+func findDuplicateVol(vols kobject.Volumes, volumes []kobject.Volumes) bool {
+	for _, v := range volumes {
+		if vols.MountPath == v.MountPath {
+			return true
+		}
+	}
+	return false
+}
+
+// returns all volumes associated with service, if `volumes_from` key is used, we have to retrieve volumes from the services which are mentioned there. Hence, recursive function is used here.
+func retrieveVolume(svcName string, komposeObject kobject.KomposeObject) (volume []kobject.Volumes) {
+	// if volumes-from key is present
+	if komposeObject.ServiceConfigs[svcName].VolumesFrom != nil {
+		// iterating over services from `volumes-from`
+		for _, depSvc := range komposeObject.ServiceConfigs[svcName].VolumesFrom {
+			// recursive call for retrieving volumes of services from `volumes-from`
+			dVols := retrieveVolume(depSvc, komposeObject)
+			var cVols []kobject.Volumes
+			// creating array of volumes of current service(svcName)
+			for _, v := range komposeObject.ServiceConfigs[svcName].VolList {
+				vol := kobject.Volumes{
+					SvcName:   svcName,
+					MountPath: v,
+				}
```

Comment:
You can use `transformer.ParseVolume` here, then you won't need to call it later in the final volume.